Initium-Odp/src/com/universeprojects/miniup/server/aspects/AspectSlotted.java

```diff
@@ -1,40 +1,184 @@
 package com.universeprojects.miniup.server.aspects;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.CommonChecks;
 import com.universeprojects.miniup.server.InitiumObject;
 import com.universeprojects.miniup.server.ItemAspect;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.UserRequestIncompleteException;
+import com.universeprojects.miniup.server.ItemAspect.ItemPopupEntry;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
 
-public class AspectSlotted extends ItemAspect
-{
+/**
+ * 
+ * @author Evan
+ *
+ */
+public class AspectSlotted extends ItemAspect {
 
-	public AspectSlotted(InitiumObject object)
-	{
+	public AspectSlotted(InitiumObject object) {
 		super(object);
 	}
 
+	//Here, we need to properly set the link. use aspectpet as a base.
 	@Override
 	public List<ItemPopupEntry> getItemPopupEntries(CachedEntity currentCharacter)
 	{
-		return null;
+		if (CommonChecks.checkItemIsAccessible(object.getEntity(), currentCharacter) && hasOpenSlots()) {
+			ItemPopupEntry ipe = new ItemPopupEntry("Insert an item into the slots", "Insert a slottable item into this item.",
+					"viewSlottableItems(event, " + object.getKey().getId() + ")");
+			return Arrays.asList(new ItemPopupEntry[]
+			{
+					ipe
+			});
+		}
+		else return null;
 	}
 
 	@Override
-	public String getPopupTag()
-	{
+	public String getPopupTag() {
 		if (getMaxCount()>0)
 			return "Slotted";
 		else
 			return null;
 	}
 
-	
-	public long getMaxCount()
-	{
-		Long count = (Long)this.getProperty("maxCount");
+	/**
+	 * Returns the number of max slots on this aspect.
+	 * @return
+	 */
+	public long getMaxCount() {
+		Long count = (Long)getProperty("maxCount");
 		if (count==null) count = 0L;
 		return count;
 	}
 	
-}
+	/**
+	 * Returns true if this aspect has any open slots.
+	 * @return
+	 */
+	public boolean hasOpenSlots() {
+		return getMaxCount() < getOpenCount();
+	}
+	
+	/**
+	 * Returns the number of open slots on this aspect.
+	 * @return
+	 */
+	public long getOpenCount() {
+		Long max = (Long)getProperty("maxCount");
+		
+		List<EmbeddedEntity> currentlySlotted = (List<EmbeddedEntity>)getProperty("slotItems"));
```

Comment:
If you wanted to get fancy you could add this get property call to it's own method on the aspect so you don't have to cast it all the time.